src/hal/SensorInterface.h

```diff
@@ -1,9 +1,14 @@
 #ifndef SENSOR_H
 #define SENSOR_H
 
+template <typename SensorReturnType, typename SensorInputType>
+
 class SensorInterface {
```

Comment:
Add the interface class (SensorInterface ) and the "concrete" classes (AngleSensor and AngleSensorProto) into a namespace called `Hal`, we'll do the same for the actuators.